--- a/finalforeach/cosmicreach/entities/MobSpawner.java
+++ b/finalforeach/cosmicreach/entities/MobSpawner.java
@@ -87,9 +_,12 @@
 	}
 
 	protected void spawnMob(Zone zone) {
+
 		String s = spawningMobEntityTypeIds.random();
 		Entity entity = EntityCreator.get(s);
 		entity.setPosition(this.tmpSpawnPackLoc);
+		if(com.github.puzzle.paradox.api.Paradox.getInstance().getEventBus().post(new com.github.puzzle.paradox.api.events.EntityEvents.OnEntitySpawn(entity.getParadoxEntity())).isCanceled())
+			return;
 		zone.addEntity(entity);
 		this.numberOfMobs++;
 	}
