--- a/finalforeach/cosmicreach/networking/netty/NettyServer.java
+++ b/finalforeach/cosmicreach/networking/netty/NettyServer.java
@@ -4,7 +_,9 @@
 import com.badlogic.gdx.utils.ObjectMap;
 import finalforeach.cosmicreach.accounts.Account;
 import finalforeach.cosmicreach.entities.player.Player;
+import finalforeach.cosmicreach.accounts.Account;
 import finalforeach.cosmicreach.networking.common.NetworkIdentity;
+import finalforeach.cosmicreach.networking.netty.packets.DisconnectPacket;
 import finalforeach.cosmicreach.networking.server.ServerIdentity;
 import finalforeach.cosmicreach.networking.server.ServerSingletons;
 import io.netty.bootstrap.ServerBootstrap;
@@ -16,12 +_,21 @@
 import io.netty.channel.nio.NioEventLoopGroup;
 import io.netty.channel.socket.SocketChannel;
 import io.netty.channel.socket.nio.NioServerSocketChannel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.util.WeakHashMap;
+
+import static finalforeach.cosmicreach.networking.server.ServerSingletons.getAccount;
 
 public class NettyServer {
 	private int port;
+
 	Array<ServerIdentity> connections = new Array<>();
-	ObjectMap<ChannelHandlerContext, ServerIdentity> contextToIdentity = new ObjectMap<>();
+	public ObjectMap<ChannelHandlerContext, ServerIdentity> contextToIdentity = new ObjectMap<>();
 
+    private final Logger LOGGER = LoggerFactory.getLogger("Cosmic Reach | Server");
 	public NettyServer(int port) {
 		this.port = port;
 	}
@@ -57,14 +_,18 @@
 		packet.setupAndSendToAllExcept(this.connections, identity);
 	}
 
+
 	public void addContext(ChannelHandlerContext ctx) {
-		ServerIdentity serveridentity = new ServerIdentity(this, ctx);
-		this.connections.add(serveridentity);
-		this.contextToIdentity.put(ctx, serveridentity);
+		ServerIdentity id = new ServerIdentity(this, ctx);
+		this.connections.add(id);
+		this.contextToIdentity.put(ctx, id);
 	}
 
 	public void removeContext(ChannelHandlerContext ctx) {
 		ServerIdentity serveridentity = this.contextToIdentity.get(ctx);
+		var addy = ((InetSocketAddress)ctx.channel().remoteAddress());
+		if(!serveridentity.waskicked)
+			LOGGER.info( "Player uid '{}' left | {}:{}", getAccount(ctx).getUniqueId(),addy.getAddress().getHostAddress() ,addy.getPort());
 		this.connections.removeValue(serveridentity, true);
 		this.contextToIdentity.remove(ctx);
 		Player player = ServerSingletons.getPlayer(serveridentity);
@@ -76,6 +_,7 @@
 	public int getPort() {
 		return this.port;
 	}
+
 
 	public Account getAccount(ChannelHandlerContext ctx) {
 		ServerIdentity serveridentity = this.contextToIdentity.get(ctx);
