--- a/finalforeach/cosmicreach/networking/netty/NettyPacketHandler.java
+++ b/finalforeach/cosmicreach/networking/netty/NettyPacketHandler.java
@@ -1,9 +_,11 @@
 package finalforeach.cosmicreach.networking.netty;
 
 import com.badlogic.gdx.utils.Array;
+import com.github.puzzle.paradox.game.event.PacketEvents;
 import finalforeach.cosmicreach.GameSingletons;
 import finalforeach.cosmicreach.networking.GamePacket;
 import finalforeach.cosmicreach.networking.NetworkIdentity;
+import finalforeach.cosmicreach.networking.packets.entities.DropItemPacket;
 import finalforeach.cosmicreach.networking.packets.meta.TransactionPacket;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
@@ -20,17 +_,28 @@
 			bytebuf = (ByteBuf)msg;
 			short short1 = bytebuf.readShort();
 			GamePacket gamepacket = GamePacket.getPacket(short1);
+			if(gamepacket == null)
+				return;
 			gamepacket.receive(bytebuf);
 			NetworkIdentity networkidentity = this.getIdentity(ctx);
+//			System.out.println("Packet: " + short1);
+			if(gamepacket instanceof DropItemPacket){
+				System.out.println("hi form drop packet");
+			}
 			if (TransactionPacket.keepBundling && !(gamepacket instanceof TransactionPacket)) {
 				this.bundledPackets.add(gamepacket);
 			} else {
-				gamepacket.handle(networkidentity, ctx);
+				PacketEvents.ON_PACKET_HANDLE_EVENT.invoker().onPacketHandle(gamepacket,networkidentity,ctx);
+				if(!gamepacket.shouldGameIgnore)
+					gamepacket.handle(networkidentity, ctx);
 			}
 
 			if (!this.bundledPackets.isEmpty() && !TransactionPacket.keepBundling) {
+				PacketEvents.ON_BUNDLE_PACKET_HANDLE_EVENT.invoker().onPacketBundleHandle(bundledPackets,networkidentity,ctx);
+
 				for (GamePacket gamepacket1 : this.bundledPackets) {
-					gamepacket1.handle(networkidentity, ctx);
+					if(!gamepacket1.shouldGameIgnore)
+						gamepacket1.handle(networkidentity, ctx);
 				}
 
 				this.bundledPackets.clear();
