--- a/finalforeach/cosmicreach/networking/netty/ServerHandler.java
+++ b/finalforeach/cosmicreach/networking/netty/ServerHandler.java
@@ -3,10 +_,10 @@
 import finalforeach.cosmicreach.networking.common.NetworkIdentity;
 import finalforeach.cosmicreach.networking.netty.packets.meta.DisconnectPacket;
 import finalforeach.cosmicreach.networking.netty.packets.meta.ProtocolSyncPacket;
-import finalforeach.cosmicreach.util.logging.Logger;
 import io.netty.channel.ChannelHandlerContext;
 
-public class ServerHandler extends NettyPacketHandler {
+public class ServerHandler extends com.github.puzzle.paradox.game.server.packets.ParadoxNettyPacketHandler {
+
 	NettyServer nettyServer;
 
 	public ServerHandler(NettyServer nettyServer) {
@@ -15,17 +_,18 @@
 
 	@Override
 	public void handlerAdded(ChannelHandlerContext ctx) {
-		Logger.info("Handler added");
+      	LOGGER.info("Handler added");
 		this.nettyServer.addContext(ctx);
 		new ProtocolSyncPacket().setupAndSend(ctx);
 	}
 
 	@Override
 	public void handlerRemoved(ChannelHandlerContext ctx) {
-		Logger.info("Handler removed");
 		this.nettyServer.broadcast(new DisconnectPacket(this.nettyServer.getAccount(ctx)));
+      	LOGGER.info("Handler removed");
 		this.nettyServer.removeContext(ctx);
 	}
+
 
 	@Override
 	public NetworkIdentity getIdentity(ChannelHandlerContext ctx) {
