--- a/finalforeach/cosmicreach/chat/commands/Command.java
+++ b/finalforeach/cosmicreach/chat/commands/Command.java
@@ -22,7 +_,7 @@
 import java.util.Map.Entry;
 import java.util.function.Supplier;
 
-public abstract class Command {
+public abstract class Command extends com.github.puzzle.paradox.game.command.ParadoxCommand {
 	private static final HashMap<String, Supplier<Command>> REGISTERED_COMMANDS = new HashMap<>();
 	private static final HashMap<String, Supplier<Command>> REGISTERED_ALIAS_COMMANDS = new HashMap<>();
 	private static final HashMap<Supplier<Command>, Array<String>> REGISTERED_COMMAND_ALIAS_LIST = new HashMap<>();
@@ -34,6 +_,8 @@
 	private String[] args;
 
 	public static void registerCommand(Supplier<Command> commandSupplier, String commandName, String... aliases) {
+		com.github.puzzle.paradox.game.command.Commands.registerVanillaCommands(commandSupplier,commandName,aliases);
+		/*
 		if (commandName == null) {
 			throw new IllegalArgumentException("Command cannot be null.");
 		} else if (commandSupplier == null) {
@@ -69,6 +_,7 @@
 				}
 			}
 		}
+		 */
 	}
 
 	public void run(IChat chat) {
