--- a/finalforeach/cosmicreach/entities/player/Player.java
+++ b/finalforeach/cosmicreach/entities/player/Player.java
@@ -5,6 +_,7 @@
 import com.badlogic.gdx.utils.Json;
 import com.badlogic.gdx.utils.JsonValue;
 import com.badlogic.gdx.utils.Json.Serializable;
+import com.github.puzzle.paradox.game.player.InternalParadoxPlayer;
 import finalforeach.cosmicreach.GameSingletons;
 import finalforeach.cosmicreach.accounts.Account;
 import finalforeach.cosmicreach.blocks.BlockState;
@@ -20,7 +_,7 @@
 import finalforeach.cosmicreach.world.World;
 import finalforeach.cosmicreach.world.Zone;
 
-public class Player implements Serializable, ISlotContainerParent {
+public class Player extends InternalParadoxPlayer implements Serializable, ISlotContainerParent {
 	public Gamemode gamemode = Gamemode.SURVIVAL;
 	public String username = "localPlayer";
 	public String zoneId;
