--- a/finalforeach/cosmicreach/GameSingletons.java
+++ b/finalforeach/cosmicreach/GameSingletons.java
@@ -29,6 +_,8 @@
 import finalforeach.cosmicreach.world.IZoneLoader;
 import finalforeach.cosmicreach.world.World;
 import finalforeach.cosmicreach.world.Zone;
+import io.netty.channel.ChannelHandlerContext;
+
 import java.util.HashMap;
 import java.util.WeakHashMap;
 import java.util.function.Function;
