--- a/finalforeach/cosmicreach/RuntimeInfo.java
+++ b/finalforeach/cosmicreach/RuntimeInfo.java
@@ -12,7 +_,7 @@
 	public static final boolean isWindows = osName.contains("windows");
 	public static final String version = Gdx.files.internal("build_assets/version.txt").readString();
 	public static boolean GL30Supported = true;
-	public static String gameVariant = "Vanilla";
+	public static String gameVariant = "Paradox";
 	private static DecimalFormat decimalFormat = new DecimalFormat("#.###");
 	private static StringBuilder sb = new StringBuilder();
 	private static final IntBuffer glInt = BufferUtils.newIntBuffer(1);
