--- a/finalforeach/cosmicreach/ZoneLoader.java
+++ b/finalforeach/cosmicreach/ZoneLoader.java
@@ -8,6 +_,7 @@
 import com.badlogic.gdx.utils.ObjectMap;
 import com.badlogic.gdx.utils.ObjectSet;
 import com.badlogic.gdx.utils.Queue;
+import com.github.puzzle.paradox.game.command.console.Save;
 import finalforeach.cosmicreach.entities.player.Player;
 import finalforeach.cosmicreach.io.ChunkLoader;
 import finalforeach.cosmicreach.io.ChunkSaver;
@@ -17,12 +_,7 @@
 import finalforeach.cosmicreach.networking.server.ServerSingletons;
 import finalforeach.cosmicreach.util.ArrayUtils;
 import finalforeach.cosmicreach.util.Point3DMap;
-import finalforeach.cosmicreach.world.Chunk;
-import finalforeach.cosmicreach.world.EntityChunk;
-import finalforeach.cosmicreach.world.EntityRegion;
-import finalforeach.cosmicreach.world.Region;
-import finalforeach.cosmicreach.world.World;
-import finalforeach.cosmicreach.world.Zone;
+import finalforeach.cosmicreach.world.*;
 import finalforeach.cosmicreach.worldgen.ChunkColumn;
 
 public class ZoneLoader {
@@ -191,7 +_,6 @@
 		int j = i;
 		int k = MathUtils.clamp(i, 6, 12);
 		if (!this.spawnReady) {
-			k = 1;
 		}
 
 		if (player.isLoading()) {
@@ -473,11 +_,12 @@
 		}
 
 		long k = System.currentTimeMillis();
-		if (this.isSaveRequested()) {
+		if (this.isSaveRequested() || Save.isSaveRequested()) {
 			this.zone.forEachEntityRegion(r -> r.save(this.zone));
 			ChunkSaver.saveWorld(world);
 			this.lastSaveMillis = k;
 			this.requestSave = false;
+			Save.shouldSave = false;
 		} else if (!this.needToGenDifferentChunks(this.zone, this.localGenRadiusInChunks) && !this.hasChunkColumnsToGenerate()) {
 			if (!this.spawnReady && GameSingletons.isHost) {
 				this.readyToPlay = false;
