--- a/finalforeach/cosmicreach/blockevents/BlockEventTrigger.java
+++ b/finalforeach/cosmicreach/blockevents/BlockEventTrigger.java
@@ -5,8 +_,11 @@
 import com.badlogic.gdx.utils.Json.Serializable;
 import finalforeach.cosmicreach.blockevents.actions.IBlockAction;
 
+import static com.github.puzzle.paradox.game.event.BlockEventEvents.ON_BLOCK_EVENT_ACT;
+
 public class BlockEventTrigger implements Serializable {
 	private IBlockAction action;
+	public boolean shouldAct = true;
 
 	@Override
 	public void write(Json json) {
@@ -29,8 +_,12 @@
 	}
 
 	public void act(BlockEventArgs args) {
+
+		ON_BLOCK_EVENT_ACT.invoker().onBlockEvenAct(args);
+
 		try {
-			this.action.act(args);
+			if(shouldAct)
+				this.action.act(args);
 		} catch (Exception exception) {
 			if (this.action != null) {
 				throw new RuntimeException("Error while triggering block event action " + this.action.getActionId() + " for block: " + args.srcBlockState, exception);
