--- a/finalforeach/cosmicreach/server/ServerLauncher.java
+++ b/finalforeach/cosmicreach/server/ServerLauncher.java
@@ -12,9 +_,13 @@
 import finalforeach.cosmicreach.world.World;
 import finalforeach.cosmicreach.worldgen.ZoneGenerator;
 import java.io.File;
+import org.slf4j.LoggerFactory;
 
 public class ServerLauncher {
-   public static void main(String[] args) throws Exception {
+    private static boolean isRunning = false;
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger("Cosmic Reach");
+
+    public static void main(String[] args) throws Exception {
       if (!StartupHelper.startNewJvmIfRequired()) {
          if (Gdx.files == null) {
             Gdx.files = new HeadlessFiles();
@@ -30,22 +_,31 @@
                new File(SaveLocation.saveLocationOverride).mkdirs();
             }
          }
-
+      
          ServerSingletons.create();
          String s2 = "server-world-1";
          GameSingletons.world = ChunkLoader.loadWorld(s2);
          if (GameSingletons.world == null) {
-            Logger.info("Creating new world.");
+         LOGGER.info("Creating new world.");
             String s3 = "base:earth";
             GameSingletons.world = World.createNew(s2, null, s3, ZoneGenerator.getZoneGenerator(s3));
          } else {
-            Logger.info("Loaded world.");
+         LOGGER.info("Loaded world.");
             WorldLoaders.INSTANCE.addZoneLoader(GameSingletons.world.getDefaultZone());
          }
 
          NettyServer nettyserver = new NettyServer();
          ServerSingletons.server = nettyserver;
+         isRunning = true;
+         ServerSingletons.puzzle.initConsoleListener();
+         ServerSingletons.puzzle.init();
          nettyserver.run();
+         isRunning = false;
+
       }
+   }
+
+   public static boolean isRunning() {
+      return isRunning;
    }
 }
