--- a/finalforeach/cosmicreach/networking/common/NetworkIdentity.java
+++ b/finalforeach/cosmicreach/networking/common/NetworkIdentity.java
@@ -5,7 +_,7 @@
 import finalforeach.cosmicreach.networking.netty.GamePacket;
 import finalforeach.cosmicreach.world.Zone;
 
-public abstract class NetworkIdentity {
+public abstract class NetworkIdentity extends com.github.puzzle.paradox.game.server.ParadoxNetworkIdentity {
 	final NetworkSide side;
 
 	public NetworkIdentity(NetworkSide side) {
