--- a/finalforeach/cosmicreach/savelib/crbin/CRBSerialized.java
+++ b/finalforeach/cosmicreach/savelib/crbin/CRBSerialized.java
@@ -1,11 +_,13 @@
 package finalforeach.cosmicreach.savelib.crbin;
 
-import java.lang.annotation.ElementType;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
 import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
-@Retention(RetentionPolicy.RUNTIME)
-@Target({ElementType.FIELD})
+@Retention(RUNTIME)
+@Target(FIELD)
 public @interface CRBSerialized {
+
 }
