--- a/finalforeach/cosmicreach/savelib/crbin/ICRBinSerializable.java
+++ b/finalforeach/cosmicreach/savelib/crbin/ICRBinSerializable.java
@@ -1,7 +_,7 @@
 package finalforeach.cosmicreach.savelib.crbin;
 
-public interface ICRBinSerializable {
-	void read(CRBinDeserializer var1);
-
-	void write(CRBinSerializer var1);
+public interface ICRBinSerializable
+{
+	void read(CRBinDeserializer deserial);
+	void write(CRBinSerializer serial);
 }
