--- a/finalforeach/cosmicreach/worldgen/ZoneGenerator.java
+++ b/finalforeach/cosmicreach/worldgen/ZoneGenerator.java
@@ -13,7 +_,7 @@
 import java.util.Random;
 
 public abstract class ZoneGenerator {
-	private static final Object REGISTER_ZONE_LOCK = new Object();
+	public static final Object REGISTER_ZONE_LOCK = new Object();
 	private static HashMap<String, Class<? extends ZoneGenerator>> allZoneGenerators;
 	private static Array<String> allZoneIds = new Array<>(String.class);
 	private static HashMap<String, String> zoneIdsAndNames = new HashMap<>();
