--- a/finalforeach/cosmicreach/savelib/utils/DynamicArrays.java
+++ b/finalforeach/cosmicreach/savelib/utils/DynamicArrays.java
@@ -2,18 +_,22 @@
 
 import finalforeach.cosmicreach.savelib.IByteArray;
 
-public class DynamicArrays {
+public class DynamicArrays
+{
 	public static IDynamicArrayInstantiator instantiator;
 
-	public static <E> IDynamicArray<E> getNew(Class<E> clazz) {
+	public static <E> IDynamicArray<E> getNew(Class<E> clazz)
+	{
 		return instantiator.create(clazz);
 	}
-
-	public static <E> IDynamicArray<E> getNew(Class<E> clazz, int initialCapacity) {
+	
+	public static <E> IDynamicArray<E> getNew(Class<E> clazz, int initialCapacity)
+	{
 		return instantiator.create(clazz, initialCapacity);
 	}
 
-	public static IByteArray getNewByteArray() {
+	public static IByteArray getNewByteArray() 
+	{
 		return instantiator.createByteArray();
 	}
 }
