--- a/finalforeach/cosmicreach/savelib/utils/IDynamicArrayInstantiator.java
+++ b/finalforeach/cosmicreach/savelib/utils/IDynamicArrayInstantiator.java
@@ -2,10 +_,9 @@
 
 import finalforeach.cosmicreach.savelib.IByteArray;
 
-public interface IDynamicArrayInstantiator {
-	<E> IDynamicArray<E> create(Class<E> var1);
-
-	<E> IDynamicArray<E> create(Class<E> var1, int var2);
-
+public interface IDynamicArrayInstantiator
+{
+	<E> IDynamicArray<E> create(Class<E> clazz);
+	<E> IDynamicArray<E> create(Class<E> clazz, int initialCapacity);
 	IByteArray createByteArray();
 }
