--- a/finalforeach/cosmicreach/savelib/utils/IObjectIntMap.java
+++ b/finalforeach/cosmicreach/savelib/utils/IObjectIntMap.java
@@ -1,7 +_,8 @@
 package finalforeach.cosmicreach.savelib.utils;
 
-public interface IObjectIntMap<K> {
-	int get(K var1, int var2);
+public interface IObjectIntMap<K> 
+{
+	int get(K key, int defaultValue);
 
-	void put(K var1, int var2);
+	void put(K key, int value);
 }
