--- a/finalforeach/cosmicreach/savelib/utils/IObjectMapInstantiator.java
+++ b/finalforeach/cosmicreach/savelib/utils/IObjectMapInstantiator.java
@@ -1,13 +_,15 @@
 package finalforeach.cosmicreach.savelib.utils;
 
-public interface IObjectMapInstantiator {
-	<K, V> IObjectMap<K, V> create(IObjectMap<K, V> var1);
+public interface IObjectMapInstantiator 
+{
+	<K, V> IObjectMap<K, V> create(IObjectMap<K, V> srcMap);
 
 	<K, V> IObjectMap<K, V> create();
 
 	<K> IObjectIntMap<K> createObjectIntMap();
 
 	<K> IObjectLongMap<K> createObjectLongMap();
-
+	
 	<K> IObjectFloatMap<K> createObjectFloatMap();
+
 }
