--- a/finalforeach/cosmicreach/savelib/utils/TriConsumer.java
+++ b/finalforeach/cosmicreach/savelib/utils/TriConsumer.java
@@ -1,6 +_,15 @@
 package finalforeach.cosmicreach.savelib.utils;
 
 @FunctionalInterface
-public interface TriConsumer<A, B, C> {
-	void accept(A var1, B var2, C var3);
+public interface TriConsumer<A, B, C>
+{
+
+    /**
+     * Performs this operation on the given arguments.
+     *
+     * @param a the first input argument
+     * @param b the second input argument
+     * @param c the second input argument
+     */
+    void accept(A a, B b, C c);
 }
