--- a/finalforeach/cosmicreach/networking/netty/packets/blocks/InteractBlockPacket.java
+++ b/finalforeach/cosmicreach/networking/netty/packets/blocks/InteractBlockPacket.java
@@ -57,7 +_,8 @@
 					blockeventargs.zone = this.zone;
 					blockeventargs.blockPos = this.blockPos;
 					blockeventargs.srcIdentity = identity;
-
+					if(!com.github.puzzle.paradox.game.player.PlayerChecks.BlockRangeCheck(identity,blockPos))
+						return;
 					for (int i = 0; i < ablockeventtrigger.length; i++) {
 						ablockeventtrigger[i].act(blockeventargs);
 					}
