--- a/finalforeach/cosmicreach/networking/netty/packets/blocks/PlaceBlockPacket.java
+++ b/finalforeach/cosmicreach/networking/netty/packets/blocks/PlaceBlockPacket.java
@@ -1,5 +_,6 @@
 package finalforeach.cosmicreach.networking.netty.packets.blocks;
 
+import com.github.puzzle.paradox.game.server.ParadoxServerSettings;
 import finalforeach.cosmicreach.GameSingletons;
 import finalforeach.cosmicreach.blockevents.BlockEventArgs;
 import finalforeach.cosmicreach.blockevents.BlockEventTrigger;
@@ -12,6 +_,7 @@
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
 
+
 public class PlaceBlockPacket extends GamePacket {
 	Zone zone;
 	BlockPosition blockPos;
@@ -42,7 +_,7 @@
 
 	@Override
 	protected void handle(NetworkIdentity identity, ChannelHandlerContext ctx) {
-		if (identity.getSide() != NetworkSide.CLIENT) {
+		if (identity.getSide() != NetworkSide.CLIENT && ParadoxServerSettings.canPlaceBlock) {
 			BlockEventTrigger[] ablockeventtrigger = this.targetBlockState.getTrigger("onPlace");
 			if (ablockeventtrigger != null) {
 				BlockEventArgs blockeventargs = new BlockEventArgs();
