--- a/finalforeach/cosmicreach/networking/netty/packets/blocks/PlaceBlockPacket.java
+++ b/finalforeach/cosmicreach/networking/netty/packets/blocks/PlaceBlockPacket.java
@@ -1,5 +_,6 @@
 package finalforeach.cosmicreach.networking.netty.packets.blocks;
 
+import com.github.puzzle.paradox.game.server.ParadoxServerSettings;
 import finalforeach.cosmicreach.GameSingletons;
 import finalforeach.cosmicreach.blockevents.BlockEventTrigger;
 import finalforeach.cosmicreach.blocks.BlockPosition;
@@ -10,6 +_,7 @@
 import finalforeach.cosmicreach.world.Zone;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
+
 import java.util.HashMap;
 import java.util.Map;
 
@@ -43,7 +_,7 @@
 
 	@Override
 	protected void handle(NetworkIdentity identity, ChannelHandlerContext ctx) {
-		if (identity.getSide() != NetworkSide.CLIENT) {
+		if (identity.getSide() != NetworkSide.CLIENT && ParadoxServerSettings.canPlaceBlock) {
 			Map<String, Object> map = new HashMap<>();
 			map.put("blockPos", this.blockPos);
 			map.put("targetBlockState", this.targetBlockState);
