--- a/finalforeach/cosmicreach/networking/server/ServerSingletons.java
+++ b/finalforeach/cosmicreach/networking/server/ServerSingletons.java
@@ -26,6 +_,7 @@
 
 public class ServerSingletons {
 	public static NettyServer server;
+  	public static com.github.puzzle.paradox.core.PuzzlePL puzzle;
 	private static ObjectMap<NetworkIdentity, Account> netIdsToAccounts = new ObjectMap<>();
 	private static ObjectMap<Account, NetworkIdentity> accountsToNetIds = new ObjectMap<>();
 	private static ObjectMap<String, Account> uniqueIdsToAccounts = new ObjectMap<>();
@@ -107,6 +_,7 @@
 
 			world.currentWorldTick++;
 		});
+		puzzle = new com.github.puzzle.paradox.core.PuzzlePL();
 		registerBlockEntityOpeners();
 		GameSingletons.postCreate();
 	}
@@ -132,6 +_,9 @@
 
 	public static Player getPlayer(NetworkIdentity identity) {
 		return GameSingletons.getPlayerFromUniqueId(getAccount(identity).getUniqueId());
+   }
+   public static ServerIdentity getIdentityByAccount(Account account) {
+      return (ServerIdentity)accountsToNetIds.get(account);
 	}
 
 	public static ServerIdentity getConnection(Player player) {
