--- a/finalforeach/cosmicreach/networking/server/ServerSingletons.java
+++ b/finalforeach/cosmicreach/networking/server/ServerSingletons.java
@@ -2,7 +_,6 @@
 
 import com.badlogic.gdx.math.Vector3;
 import com.badlogic.gdx.utils.ObjectMap;
-import finalforeach.cosmicreach.BlockEntityScreenInfo;
 import finalforeach.cosmicreach.GameSingletons;
 import finalforeach.cosmicreach.Threads;
 import finalforeach.cosmicreach.TickRunner;
@@ -36,6 +_,8 @@
 
 public class ServerSingletons {
 	public static NettyServer SERVER;
+
+  	public static com.github.puzzle.paradox.core.PuzzlePL puzzle;
 	private static ObjectMap<NetworkIdentity, Account> netIdsToAccounts = new ObjectMap<>();
 	private static ObjectMap<Account, NetworkIdentity> accountsToNetIds = new ObjectMap<>();
 	private static ObjectMap<String, Account> uniqueIdsToAccounts = new ObjectMap<>();
@@ -136,6 +_,7 @@
 			World world = GameSingletons.world;
 
 			for (Zone zone : world.getZones()) {
+
 				zone.update(deltaTime);
 			}
 
@@ -144,6 +_,7 @@
 				SERVER.broadcastToAll(new EndTickPacket(world.currentWorldTick));
 			}
 		});
+		puzzle = new com.github.puzzle.paradox.core.PuzzlePL();
 		registerBlockEntityOpeners();
 		GameSingletons.postCreate();
 	}
@@ -168,15 +_,17 @@
 			serveridentity.send(new BlockEntityScreenPacket(blockentity));
 		});
 	}
-
 	public static Account getAccount(NetworkIdentity identity) {
 		return identity == null ? null : netIdsToAccounts.get(identity);
 	}
-
 	public static Player getPlayer(NetworkIdentity identity) {
 		Account account = getAccount(identity);
 		return account == null ? null : GameSingletons.getPlayerFromUniqueId(account.getUniqueId());
+    }
+    public static ServerIdentity getIdentityByAccount(Account account) {
+      return (ServerIdentity)accountsToNetIds.get(account);
 	}
+
 
 	public static ServerIdentity getConnection(Player player) {
 		Account account = GameSingletons.getAccountFromPlayer(player);
