--- a/finalforeach/cosmicreach/networking/server/ServerZoneLoader.java
+++ b/finalforeach/cosmicreach/networking/server/ServerZoneLoader.java
@@ -2,13 +_,13 @@
 
 import com.badlogic.gdx.utils.Array;
 import finalforeach.cosmicreach.ZoneLoader;
-import finalforeach.cosmicreach.entities.player.Player;
 import finalforeach.cosmicreach.world.IZoneLoader;
 import finalforeach.cosmicreach.world.Zone;
 import finalforeach.cosmicreach.worldgen.ChunkColumn;
 
 public class ServerZoneLoader implements IZoneLoader {
 	public int serverLoadDistance = 10;
+	public static ServerZoneLoader INSTANCE;
 	private ZoneLoader worldLoader = new ZoneLoader();
 	public final Zone zone;
 	boolean exit;
@@ -29,6 +_,7 @@
 
 	public ServerZoneLoader(Zone zone) {
 		this.zone = zone;
+		INSTANCE = this;
 		this.worldLoader.lastRenderDist = this.serverLoadDistance;
 		this.worldLoader.lastSaveMillis = System.currentTimeMillis();
 	}
