--- a/finalforeach/cosmicreach/networking/server/ServerIdentity.java
+++ b/finalforeach/cosmicreach/networking/server/ServerIdentity.java
@@ -6,11 +_,9 @@
 import finalforeach.cosmicreach.networking.netty.GamePacket;
 import finalforeach.cosmicreach.networking.netty.NettyServer;
 import io.netty.channel.ChannelHandlerContext;
-
 public class ServerIdentity extends NetworkIdentity {
 	public final NettyServer server;
 	public final ChannelHandlerContext ctx;
-
 	public ServerIdentity(NettyServer nettyServer, ChannelHandlerContext ctx) {
 		super(NetworkSide.SERVER);
 		this.server = nettyServer;
