--- a/finalforeach/cosmicreach/accounts/AccountOffline.java
+++ b/finalforeach/cosmicreach/accounts/AccountOffline.java
@@ -7,14 +_,18 @@
 public class AccountOffline extends Account {
 	private String displayName = "Player";
 
+
+	public static int accountIndex = 0;
 	public AccountOffline() {
-		this.setUsername("localPlayer");
+		this.setUsername("Player_" + accountIndex );
+		accountIndex++;
 		this.setUniqueId(Long.valueOf(MathUtils.random(Long.MAX_VALUE)));
+		if(isAllowed()) isValid = false;
 	}
 
 	@Override
 	public String getPrefix() {
-		return "offline";
+		return this.displayname;
 	}
 
 	@Override
@@ -29,7 +_,7 @@
 
 	@Override
 	public boolean isAllowed() {
-		return ServerSettings.OFFLINE_MODE.getValue();
+		return com.github.puzzle.paradox.game.server.ParadoxServerSettings.isOffline;
 	}
 
 	public void setDisplayName(String offlineName) {
