--- a/finalforeach/cosmicreach/accounts/AccountOffline.java
+++ b/finalforeach/cosmicreach/accounts/AccountOffline.java
@@ -3,9 +_,13 @@
 import com.badlogic.gdx.math.MathUtils;
 
 public class AccountOffline extends Account {
+
+	public static int accountIndex = 0;
 	public AccountOffline() {
-		this.setUsername("localPlayer");
+		this.setUsername("Player_" + accountIndex );
+		accountIndex++;
 		this.setUniqueId(Long.valueOf(MathUtils.random(Long.MAX_VALUE)));
+		if(com.github.puzzle.paradox.game.server.ParadoxServerSettings.isOffline) isValid = false;
 	}
 
 	@Override
@@ -15,7 +_,7 @@
 
 	@Override
 	public String getDisplayName() {
-		return "Player";
+		return this.displayname;
 	}
 
 	@Override
