--- a/finalforeach/cosmicreach/accounts/AccountOffline.java
+++ b/finalforeach/cosmicreach/accounts/AccountOffline.java
@@ -3,8 +_,11 @@
 import com.badlogic.gdx.math.MathUtils;
 
 public class AccountOffline extends Account {
+
+	static int accountIndex = 0;
 	public AccountOffline() {
-		this.setUsername("localPlayer");
+		this.setUsername("Player_" + accountIndex );
+		accountIndex++;
 		this.setUniqueId(Long.valueOf(MathUtils.random(Long.MAX_VALUE)));
 	}
 
@@ -15,6 +_,6 @@
 
 	@Override
 	public String getDisplayName() {
-		return "Player";
+		return this.displayname;
 	}
 }
