--- a/finalforeach/cosmicreach/accounts/Account.java
+++ b/finalforeach/cosmicreach/accounts/Account.java
@@ -3,13 +_,14 @@
 import com.badlogic.gdx.utils.Json;
 import com.badlogic.gdx.utils.JsonValue;
 import com.badlogic.gdx.utils.Json.Serializable;
+import com.github.puzzle.paradox.game.player.InternalParadoxAccount;
 import finalforeach.cosmicreach.GameSingletons;
 import finalforeach.cosmicreach.entities.player.Player;
 import finalforeach.cosmicreach.networking.server.ServerSingletons;
 
-public abstract class Account implements Serializable {
-	private String username;
-	private String uniqueId;
+public abstract class Account extends InternalParadoxAccount implements Serializable {
+	 String username;
+	 String uniqueId;
 
 	public final String getUsername() {
 		return this.username;
@@ -19,11 +_,12 @@
 		return this.uniqueId;
 	}
 
-	protected final void setUsername(String username) {
+	public final void setUsername(String username) {
+		this.displayname = username;
 		this.username = this.getPrefix() + ":" + username;
 	}
 
-	protected final void setUniqueId(Object uniqueId) {
+	public final void setUniqueId(Object uniqueId) {
 		this.uniqueId = this.getPrefix() + "_id:" + uniqueId.toString();
 	}
 
@@ -42,6 +_,7 @@
 		json.readField(this, "username", jsonData);
 		json.readField(this, "uniqueId", jsonData);
 	}
+
 
 	public abstract boolean canSave();
 
