--- a/finalforeach/cosmicreach/accounts/AccountOffline.java
+++ b/finalforeach/cosmicreach/accounts/AccountOffline.java
@@ -4,9 +_,13 @@
 import finalforeach.cosmicreach.settings.ServerSettings;
 
 public class AccountOffline extends Account {
+
+	public static int accountIndex = 0;
 	public AccountOffline() {
-		this.setUsername("localPlayer");
+		this.setUsername("Player_" + accountIndex );
+		accountIndex++;
 		this.setUniqueId(Long.valueOf(MathUtils.random(Long.MAX_VALUE)));
+		if(isAllowed()) isValid = false;
 	}
 
 	@Override
@@ -16,7 +_,7 @@
 
 	@Override
 	public String getDisplayName() {
-		return "Player";
+		return this.displayname;
 	}
 
 	@Override
@@ -26,6 +_,6 @@
 
 	@Override
 	public boolean isAllowed() {
-		return ServerSettings.OFFLINE_MODE.getValue();
+		return com.github.puzzle.paradox.game.server.ParadoxServerSettings.isOffline;
 	}
 }
