--- a/finalforeach/cosmicreach/networking/GamePacket.java
+++ b/finalforeach/cosmicreach/networking/GamePacket.java
@@ -57,6 +_,9 @@
 
 public abstract class GamePacket {
 	public static final Json json = new Json();
+	public boolean shouldGameIgnore = false;
+	private static final org.slf4j.Logger LOGGER =  org.slf4j.LoggerFactory.getLogger("Cosmic Reach | Game packet");
+	private static int vanillaPacketIndex = 0;
 	protected static final IntMap<Supplier<? extends GamePacket>> idsToPackets = new IntMap<>();
 	protected static final ObjectIntMap<Class<? extends GamePacket>> packetsToIntIds = new ObjectIntMap<>();
 	protected static final ObjectIntMap<String> packetNamesToIntIds = new ObjectIntMap<>();
@@ -104,6 +_,10 @@
 	}
 
 	public static <T extends GamePacket> void registerPacket(Class<T> packetClass) {
+		if(!packetClass.getName().startsWith("finalforeach.cosmicreach.networking")){
+			LOGGER.warn("Not registering packet: {}. Method reserved for vanilla packets. Please use registerPacket(String strId, int numId, Class<T> packetClass) instead.",packetClass.getName());
+			return;
+		}
 		String s = packetClass.getCanonicalName();
 		if (s == null) {
 			String s1 = "Packet class must have a canonical name";
@@ -136,14 +_,14 @@
 			packetsToIntIds.put(packetClass, numId);
 			packetNamesToIntIds.put(strId, numId);
 			packetNamesToClasses.put(strId, packetClass);
-			Logger.info("Registered packet ( id = " + numId + " ):" + strId);
+			LOGGER.info("Registered packet ( id = " + numId + " ):" + strId);
 		} catch (Exception exception) {
 			throw new RuntimeException(exception);
 		}
 	}
 
 	public GamePacket() {
-		this.packetID = (short)packetsToIntIds.get((Class<? extends GamePacket>)this.getClass(), 0);
+		this.packetID = (short)packetsToIntIds.get(this.getClass(), 0);
 		if (this.packetID == 0) {
 			throw new RuntimeException("Cannot send packet for " + this.getClass() + " without registering it first!");
 		}
@@ -154,7 +_,9 @@
 	public static GamePacket getPacket(short packetId) {
 		Supplier<? extends GamePacket> supplier = idsToPackets.get(packetId);
 		if (supplier == null) {
-			throw new NullPointerException("Unknown packet for id: " + packetId);
+//			throw new NullPointerException("Unknown packet for id: " + packetId);
+			LOGGER.warn("Unknown packet for id: " + packetId);
+			return null;
 		} else {
 			GamePacket gamepacket = supplier.get();
 			gamepacket.packetID = packetId;
