--- a/finalforeach/cosmicreach/networking/NetworkIdentity.java
+++ b/finalforeach/cosmicreach/networking/NetworkIdentity.java
@@ -6,7 +_,7 @@
 import finalforeach.cosmicreach.world.World;
 import finalforeach.cosmicreach.world.Zone;
 
-public abstract class NetworkIdentity {
+public abstract class NetworkIdentity extends com.github.puzzle.paradox.game.server.ParadoxNetworkIdentity {
 	final NetworkSide side;
 	final NetworkSettings settings;
 
