--- a/finalforeach/cosmicreach/networking/packets/meta/ItchSessionTokenPacket.java
+++ b/finalforeach/cosmicreach/networking/packets/meta/ItchSessionTokenPacket.java
@@ -2,6 +_,7 @@
 
 import com.badlogic.gdx.utils.JsonReader;
 import finalforeach.cosmicreach.accounts.Account;
+import com.badlogic.gdx.utils.SerializationException;
 import finalforeach.cosmicreach.accounts.AccountItch;
 import finalforeach.cosmicreach.networking.GamePacket;
 import finalforeach.cosmicreach.networking.NetworkIdentity;
@@ -12,6 +_,8 @@
 import finalforeach.cosmicreach.util.logging.Logger;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
+
+import java.io.IOException;
 import java.io.OutputStream;
 import java.net.HttpURLConnection;
 import java.net.MalformedURLException;
