--- a/finalforeach/cosmicreach/networking/packets/meta/LoginPacket.java
+++ b/finalforeach/cosmicreach/networking/packets/meta/LoginPacket.java
@@ -7,6 +_,7 @@
 import finalforeach.cosmicreach.networking.GamePacket;
 import finalforeach.cosmicreach.networking.NetworkIdentity;
 import finalforeach.cosmicreach.networking.server.ServerIdentity;
+import finalforeach.cosmicreach.networking.packets.MessagePacket;
 import finalforeach.cosmicreach.networking.server.ServerSingletons;
 import finalforeach.cosmicreach.settings.ServerSettings;
 import finalforeach.cosmicreach.util.logging.Logger;
@@ -15,6 +_,11 @@
 import io.netty.util.AttributeKey;
 import java.net.InetSocketAddress;
 
+import java.net.InetSocketAddress;
+import java.util.Objects;
+
+import static com.github.puzzle.paradox.core.PuzzlePL.SERVER_ACCOUNT;
+
 public class LoginPacket extends GamePacket {
 	private static final AttributeKey<Boolean> LOGIN_SET_ATTRIB = AttributeKey.valueOf("LOGIN_SET");
 	public Account account;
@@ -22,6 +_,8 @@
 	public LoginPacket() {
 	}
 
+	private static final org.slf4j.Logger LOGGER = org.slf4j.LoggerFactory.getLogger("Cosmic Reach | Login Packet");
+
 	public LoginPacket(Account account) {
 		this.account = account;
 	}
@@ -42,6 +_,7 @@
 			default -> throw new IllegalArgumentException("Unexpected value: " + s);
 		};
 		this.account = this.readObjectFromJson(in, oclass);
+
 	}
 
 	@Override
@@ -52,36 +_,42 @@
 	@Override
 	public void handle(NetworkIdentity identity, ChannelHandlerContext ctx) {
 		if (!identity.isClient()) {
+			com.github.puzzle.paradox.game.event.PacketEvents.ON_PLAYER_JOIN_EVENT.invoker().onPlayerJoin(account, identity, ctx);
+			var addy = ((InetSocketAddress) ctx.channel().remoteAddress());
+
 			if (ctx.channel().hasAttr(LOGIN_SET_ATTRIB)) {
 				ServerSingletons.SERVER.kick("Already recieved a login packet!", identity);
 			}
 
+			LOGGER.debug("Received login packet from " + this.account.getUsername());
 			ctx.channel().attr(LOGIN_SET_ATTRIB).set(true);
 			Logger.debug("Received login packet from " + this.account.getUsername());
 			if (ServerSettings.WHITELIST_ENABLED.isTrue() && !ServerSingletons.WHITELIST.hasAccount(this.account)) {
 				Logger.info("Account is not in the whitelist: " + this.account.getUsername());
 				ServerSingletons.SERVER.kick("You are not whitelisted on this server.", identity);
 			} else if (ServerSingletons.BAN_LIST.hasAccount(this.account)) {
-				Logger.info("Account is in the banlist: " + this.account.getUsername());
+				LOGGER.info("Account is in the banlist: " + this.account.getUsername());
 				ServerSingletons.SERVER.kick("You are banned on this server.", identity);
 			} else {
-				String s = ((InetSocketAddress)ctx.channel().remoteAddress()).getAddress().getHostAddress();
+			String s = ((InetSocketAddress) ctx.channel().remoteAddress()).getAddress().getHostAddress();
 				if (ServerSingletons.BANNED_IP_LIST.hasAddress(s)) {
-					Logger.info("IP is in the banlist: " + s);
+				LOGGER.info("IP is in the banlist: " + ctx.channel().remoteAddress());
 					ServerSingletons.SERVER.kick("You are banned on this server.", identity);
 				} else {
 					if (this.account instanceof IAuthenticatedAccount iauthenticatedaccount) {
 						iauthenticatedaccount.challengeClientFromServer(identity, ctx);
-					} else if (this.account.isAllowed() && ServerSettings.OFFLINE_MODE.isTrue()) {
-						ServerSingletons.addAccount(identity, this.account);
-						ServerSingletons.SERVER.addAuthenticatedIdentity((ServerIdentity)identity);
-						Logger.info("Is it really " + this.account.getUsername() + "? I have no choice to believe so (Offline mode)!");
+			}
+			else if (account.isAllowed() && ServerSettings.OFFLINE_MODE.isTrue()) {
+//				Logger.info("Is it really " + this.account.getUsername() + "? I have no choice to believe so (Offline mode)!");
 					} else {
-						Logger.info("Unable to authenticate " + this.account.getUsername());
+				LOGGER.info("Unable to authenticate " + this.account.getUsername());
 						ServerSingletons.SERVER.kick("Unable to authenticate your account.", identity);
 					}
 				}
 			}
+			LOGGER.info("Player uid '{}' joined | {}:{}", this.account.getUniqueId(), addy.getAddress().getHostAddress(), addy.getPort());
 		}
+
 	}
+
 }
