--- a/finalforeach/cosmicreach/networking/packets/meta/LoginPacket.java
+++ b/finalforeach/cosmicreach/networking/packets/meta/LoginPacket.java
@@ -6,17 +_,23 @@
 import finalforeach.cosmicreach.networking.GamePacket;
 import finalforeach.cosmicreach.networking.NetworkIdentity;
 import finalforeach.cosmicreach.networking.NetworkSide;
+import finalforeach.cosmicreach.networking.packets.MessagePacket;
 import finalforeach.cosmicreach.networking.server.ServerSingletons;
 import finalforeach.cosmicreach.util.logging.Logger;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
 
+import java.net.InetSocketAddress;
+import java.util.Objects;
+
+import static com.github.puzzle.paradox.core.PuzzlePL.SERVER_ACCOUNT;
+
 public class LoginPacket extends GamePacket {
 	public Account account;
 
 	public LoginPacket() {
 	}
-
+	private static final org.slf4j.Logger LOGGER = org.slf4j.LoggerFactory.getLogger("Cosmic Reach | Login Packet");
 	public LoginPacket(Account account) {
 		this.account = account;
 	}
@@ -37,14 +_,28 @@
 			default -> throw new IllegalArgumentException("Unexpected value: " + s);
 		};
 		this.account = this.readObjectFromJson(in, oclass);
+
 	}
 
 	@Override
 	public void handle(NetworkIdentity identity, ChannelHandlerContext ctx) {
 		if (identity.getSide() == NetworkSide.SERVER) {
+			com.github.puzzle.paradox.game.event.PacketEvents.ON_PLAYER_JOIN_EVENT.invoker().onPlayerJoin(account,identity,ctx);
+			var addy = ((InetSocketAddress)ctx.channel().remoteAddress());
+			if((this.account.getPrefix().equals("offline") && !com.github.puzzle.paradox.game.server.ParadoxServerSettings.isOffline) || (account.isValid && !com.github.puzzle.paradox.game.server.ParadoxServerSettings.isOffline)){
+				ctx.close();
+				identity.shouldIgnore = true;
+				System.out.println("Denied account: " + account.getUniqueId() + " was invalid or an offline account | ip: " + addy.getAddress().getHostAddress() + ":" + addy.getPort() );
+				return;
+			}
 			Logger.debug("Received login packet from " + this.account.getUsername());
 			ServerSingletons.addAccount(identity, this.account);
-			Logger.info("Is it really " + this.account.getUsername() + "? I have no choice to believe so!");
+			LOGGER.info( "Player uid '{}' joined | {}:{}",this.account.getUniqueId(),addy.getAddress().getHostAddress() ,addy.getPort());
+			if(!com.github.puzzle.paradox.game.server.ParadoxServerSettings.joinMessage.isEmpty()|| !com.github.puzzle.paradox.game.server.ParadoxServerSettings.joinMessage.isBlank()){
+				var pack = new MessagePacket(com.github.puzzle.paradox.game.server.ParadoxServerSettings.joinMessage);
+				pack.playerUniqueId = SERVER_ACCOUNT.getUniqueId();
+				ServerSingletons.SERVER.send(pack, ctx);
+			}
 		}
 	}
 }
