--- a/finalforeach/cosmicreach/networking/packets/blocks/BreakBlockPacket.java
+++ b/finalforeach/cosmicreach/networking/packets/blocks/BreakBlockPacket.java
@@ -43,7 +_,7 @@
 
 	@Override
 	public void handle(NetworkIdentity identity, ChannelHandlerContext ctx) {
-		if (identity.getSide() != NetworkSide.CLIENT) {
+		if (identity.getSide() != NetworkSide.CLIENT && com.github.puzzle.paradox.game.server.ParadoxServerSettings.canBreakBlock) {
 			if (this.blockPos.getBlockState() == this.brokenBlockState) {
 				BlockEventTrigger[] ablockeventtrigger = this.brokenBlockState.getTrigger("onBreak");
 				if (ablockeventtrigger != null) {
@@ -52,7 +_,10 @@
 					blockeventargs.zone = this.zone;
 					blockeventargs.blockPos = this.blockPos;
 					blockeventargs.setSrcIdentity(identity);
-
+					if(!com.github.puzzle.paradox.game.player.PlayerChecks.blockChecks(identity,blockPos))
+						return;
+					if(!com.github.puzzle.paradox.game.player.PlayerChecks.blockBreakTimingCheck(identity.getPlayer().getParadoxPlayer(), blockPos))
+						return;
 					for (int i = 0; i < ablockeventtrigger.length; i++) {
 						ablockeventtrigger[i].act(blockeventargs);
 					}
