--- a/finalforeach/cosmicreach/networking/packets/blocks/InteractBlockPacket.java
+++ b/finalforeach/cosmicreach/networking/packets/blocks/InteractBlockPacket.java
@@ -86,7 +_,8 @@
 					blockeventargs.zone = zone;
 					blockeventargs.blockPos = this.blockPos;
 					blockeventargs.srcIdentity = identity;
-
+					if(!com.github.puzzle.paradox.game.player.PlayerChecks.blockChecks(identity,blockPos))
+						return;
 					for (int j = 0; j < ablockeventtrigger.length; j++) {
 						ablockeventtrigger[j].act(blockeventargs);
 					}
