--- a/finalforeach/cosmicreach/networking/packets/blocks/PlaceBlockPacket.java
+++ b/finalforeach/cosmicreach/networking/packets/blocks/PlaceBlockPacket.java
@@ -43,7 +_,7 @@
 
 	@Override
 	public void handle(NetworkIdentity identity, ChannelHandlerContext ctx) {
-		if (identity.getSide() != NetworkSide.CLIENT) {
+		if (identity.getSide() != NetworkSide.CLIENT && com.github.puzzle.paradox.game.server.ParadoxServerSettings.canPlaceBlock) {
 			BlockEventTrigger[] ablockeventtrigger = this.targetBlockState.getTrigger("onPlace");
 			if (ablockeventtrigger != null) {
 				BlockEventArgs blockeventargs = new BlockEventArgs();
@@ -51,7 +_,8 @@
 				blockeventargs.zone = this.zone;
 				blockeventargs.blockPos = this.blockPos;
 				blockeventargs.srcIdentity = identity;
-
+				if(!com.github.puzzle.paradox.game.player.PlayerChecks.blockChecks(identity,blockPos))
+					return;
 				for (int i = 0; i < ablockeventtrigger.length; i++) {
 					ablockeventtrigger[i].act(blockeventargs);
 				}
