--- a/finalforeach/cosmicreach/savelib/lightdata/blocklight/IBlockLightData.java
+++ b/finalforeach/cosmicreach/savelib/lightdata/blocklight/IBlockLightData.java
@@ -3,10 +_,12 @@
 import finalforeach.cosmicreach.savelib.IChunkByteWriter;
 import finalforeach.cosmicreach.savelib.ISaveFileConstant;
 
-public interface IBlockLightData extends ISaveFileConstant {
-	short getBlockLight(int var1, int var2, int var3);
-
-	void setBlockLight(int var1, int var2, int var3, int var4, int var5, int var6);
-
-	void writeTo(IChunkByteWriter var1);
+public interface IBlockLightData extends ISaveFileConstant
+{
+	short getBlockLight(int localX, int localY, int localZ);
+
+	void setBlockLight(int lightLevelRed, int lightLevelGreen, int lightLevelBlue, int localX, int localY, int localZ);
+
+	void writeTo(IChunkByteWriter allChunksWriter);
+
 }
