--- a/finalforeach/cosmicreach/items/Item.java
+++ b/finalforeach/cosmicreach/items/Item.java
@@ -7,6 +_,7 @@
 import finalforeach.cosmicreach.util.logging.Logger;
 
 public interface Item {
+	org.slf4j.Logger LOGGER =  org.slf4j.LoggerFactory.getLogger("Cosmic Reach | Items");
 	ObjectMap<String, Item> allItems = new ObjectMap<>();
 
 	String getID();
@@ -27,7 +_,7 @@
 			throw new DuplicateIDException("Duplicate item for id: " + item.getID());
 		} else {
 			allItems.put(s, item);
-			Logger.info("Registered item id: " + item.getID());
+            LOGGER.info("Registered item id: " + item.getID());
 		}
 	}
 
