--- a/finalforeach/cosmicreach/items/Item.java
+++ b/finalforeach/cosmicreach/items/Item.java
@@ -7,10 +_,10 @@
 import finalforeach.cosmicreach.util.exceptions.DuplicateIDException;
 
 public interface Item {
+	org.slf4j.Logger LOGGER =  org.slf4j.LoggerFactory.getLogger("Cosmic Reach | Items");
 	ObjectMap<String, Item> allItems = new ObjectMap<>();
 
 	String getID();
-
 	boolean canMergeWith(Item var1);
 
 	boolean canMergeWithSwapGroup(Item var1);
@@ -27,6 +_,7 @@
 			throw new DuplicateIDException("Duplicate item for id: " + item.getID());
 		} else {
 			allItems.put(s, item);
+            LOGGER.info("Registered item id: " + item.getID());
 		}
 	}
 
