--- a/finalforeach/cosmicreach/items/ItemThing.java
+++ b/finalforeach/cosmicreach/items/ItemThing.java
@@ -1,6 +_,5 @@
 package finalforeach.cosmicreach.items;
 
-import com.badlogic.gdx.files.FileHandle;
 import com.badlogic.gdx.math.MathUtils;
 import com.badlogic.gdx.utils.JsonValue;
 import com.badlogic.gdx.utils.ObjectFloatMap;
@@ -18,6 +_,7 @@
 import finalforeach.cosmicreach.networking.server.ServerSingletons;
 import finalforeach.cosmicreach.world.BlockSetter;
 import finalforeach.cosmicreach.world.Zone;
+
 import java.util.HashMap;
 import java.util.function.Supplier;
 
