--- a/finalforeach/cosmicreach/savelib/blockdata/layers/IBlockLayer.java
+++ b/finalforeach/cosmicreach/savelib/blockdata/layers/IBlockLayer.java
@@ -1,17 +_,15 @@
 package finalforeach.cosmicreach.savelib.blockdata.layers;
 
 import finalforeach.cosmicreach.savelib.IChunkByteWriter;
+import finalforeach.cosmicreach.savelib.ISavedChunk;
 import finalforeach.cosmicreach.savelib.blockdata.LayeredBlockData;
 
-public interface IBlockLayer<T> {
-	int CHUNK_WIDTH = 16;
-	int NUM_BLOCKS_IN_LAYER = 256;
-
-	T getBlockValue(LayeredBlockData<T> var1, int var2, int var3);
-
-	void setBlockValue(LayeredBlockData<T> var1, T var2, int var3, int var4, int var5);
-
-	int getSaveFileConstant(LayeredBlockData<T> var1);
-
-	void writeTo(LayeredBlockData<T> var1, IChunkByteWriter var2);
+public interface IBlockLayer<T>
+{
+	public static final int CHUNK_WIDTH = ISavedChunk.CHUNK_WIDTH;
+	public static final int NUM_BLOCKS_IN_LAYER = CHUNK_WIDTH * CHUNK_WIDTH;
+	T getBlockValue(LayeredBlockData<T> chunkData, int localX, int localZ);
+	void setBlockValue(LayeredBlockData<T> chunkData, T blockValue, int localX, int localY, int localZ);
+	int getSaveFileConstant(LayeredBlockData<T> chunkData);
+	void writeTo(LayeredBlockData<T> chunkData, IChunkByteWriter allChunksWriter);	
 }
