--- a/finalforeach/cosmicreach/blockevents/actions/BlockActionExplode.java
+++ b/finalforeach/cosmicreach/blockevents/actions/BlockActionExplode.java
@@ -26,6 +_,8 @@
 
 	@Override
 	public void act(BlockEventArgs args) {
+		if(!com.github.puzzle.paradox.game.server.ParadoxServerSettings.doesC4Explode)
+			return;
 		Zone zone = args.zone;
 		BlockState blockstate = args.srcBlockState;
 		BlockPosition blockposition = args.blockPos;
