--- a/finalforeach/cosmicreach/networking/packets/entities/DropItemPacket.java
+++ b/finalforeach/cosmicreach/networking/packets/entities/DropItemPacket.java
@@ -40,11 +_,11 @@
 		if (!identity.isClient()) {
 			Player player = identity.getPlayer();
 
-			SlotContainer slotcontainer = (SlotContainer)(switch (this.windowId) {
+			SlotContainer slotcontainer = switch (this.windowId) {
 				case 0 -> player.inventory;
 				case 1 -> player.craftingContainer;
 				default -> throw new IllegalArgumentException("Unexpected value: " + this.windowId);
-			});
+			};
 			ItemSlot itemslot = slotcontainer.getSlot(this.itemSlotNum);
 			if (itemslot != null) {
 				ItemStack itemstack = itemslot.itemStack;
