--- a/finalforeach/cosmicreach/networking/packets/ZonePacket.java
+++ b/finalforeach/cosmicreach/networking/packets/ZonePacket.java
@@ -10,14 +_,14 @@
 import finalforeach.cosmicreach.world.Zone;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
-
 public class ZonePacket extends GamePacket {
 	public Zone zone;
 	boolean setDefault;
-
+	private static final org.slf4j.Logger LOGGER = org.slf4j.LoggerFactory.getLogger("Cosmic Reach | Zone Packet");
 	public ZonePacket() {
 	}
 
+
 	public ZonePacket(Zone zone, boolean setDefault) {
 		this();
 		this.zone = zone;
@@ -39,7 +_,7 @@
 	@Override
 	public void handle(NetworkIdentity identity, ChannelHandlerContext ctx) {
 		if (!identity.isServer()) {
-			Logger.info("ZonePacket: Got zone for " + this.zone.zoneId);
+			LOGGER.info("ZonePacket: Got zone for " + this.zone.zoneId);
 			World world = GameSingletons.world;
 			if (world == null) {
 				world = GameSingletons.world = NetworkedWorld.createNew();
